microovn/src/charm.py

```diff
@@ -92,49 +147,74 @@ def find_token(self, relation):
 
     def join_with_token(self, token):
         self.unit.status = ops.MaintenanceStatus("Joining cluster")
-        os.system("microovn cluster join {}".format(token))
+        error, _ = call_microovn_command("cluster", "join", token)
+        if error:
+            return False
         self._stored.in_cluster=True
         self.unit.status = ops.ActiveStatus("Joined cluster")
+        return True
 
     def _on_install(self, event: ops.InstallEvent):
         self.unit.status = ops.MaintenanceStatus("installing microovn snap")
-        os.system("snap install microovn --channel latest/edge")
+
+        subprocess.run(["snap", "install", "microovn", "--channel", "latest/edge"])
```

Comment:
Should we add a `check` argument here so that failure of `snap install` would be raised?